pkg/router/template/router.go

```diff
@@ -203,6 +203,19 @@ func matchPattern(pattern, s string) bool {
 	return false
 }
 
+// Generate a regular expression to match wildcard hosts (and paths if any)
+// for a [sub]domain.
+func genDomainWildcardRegexp(hostname, path string, exactPath bool) string {
+	route := &routeapi.Route{Spec: routeapi.RouteSpec{Host: hostname}}
+	host, _ := routeapi.NormalizeWildcardHost(route.Spec.Host)
```

Comment:
Wouldn't harm if we explicitly just checked the boolean value returned by NormalizeWildcardHost. Its a function exposed in the template, who knows how/where field use may call it with. If I just call it without a leading '*' for example, outside the scope of $cfg.IsWildcard check.